pkg/importer/http-datasource.go

```diff
@@ -196,7 +197,19 @@ func (hs *HTTPDataSource) GetURL() *url.URL {
 
 // GetTerminationMessage returns data to be serialized and used as the termination message of the importer.
 func (hs *HTTPDataSource) GetTerminationMessage() *common.TerminationMessage {
-	return nil
+	if pullMethod, _ := util.ParseEnvVar(common.ImporterPullMethod, false); pullMethod != string(cdiv1.RegistryPullNode) {
+		return nil
+	}
+
+	info, err := getServerInfo(hs.ctx, fmt.Sprintf("%s://%s/info", hs.endpoint.Scheme, hs.endpoint.Host))
+	if err != nil {
+		klog.Errorf("%+v", err)
+		return nil
+	}
+
+	return &common.TerminationMessage{
+		Labels: envsToLabels(info.Env),
```

Comment:
any reason to worry about `nil`s here? (and later)
from a quick look I don't think so but may deserve a unit test

Reply:
ACK, added a unit test for that.